build/building/context.md

```diff
@@ -0,0 +1,206 @@
+---
+title: Build context
+description: Learn how to use the build context to access files from your Dockerfile
+keywords: build, buildx, buildkit, context, git, tarball, stdin
+---
+
+The [`docker build`](../../engine/reference/commandline/build.md) or
+[`docker buildx build`](../../engine/reference/commandline/buildx_build.md)
+commands build Docker images from a [Dockerfile](../../engine/reference/builder.md)
+and a "context".
+
+A build's context is the set of files located at the `PATH` or `URL` specified
+as the positional argument to the build command:
+
+```console
+$ docker build [OPTIONS] PATH | URL | -
+                         ^^^^^^^^^^^^^^
+```
+
+The build process can refer to any of the files in the context. For example,
+your build can use a [`COPY` instruction](../../engine/reference/builder.md#copy)
+to reference a file in the context or a [`RUN --mount=type=bind` instruction](../../engine/reference/builder.md#run---mounttypebind)
+for better performance with [BuildKit](../buildkit/index.md). The build context
+is processed recursively. So, a `PATH` includes any subdirectories and the
+`URL` includes the repository and its submodules.
+
+## `PATH` context
+
+This example shows a build command that uses the current directory (`.`) as a
+build context:
+
+```console
+$ docker build .
+...
+#16 [internal] load build context
+#16 sha256:23ca2f94460dcbaf5b3c3edbaaa933281a4e0ea3d92fe295193e4df44dc68f85
+#16 transferring context: 13.16MB 2.2s done
+...
+```
+
+With the following Dockerfile:
+
+```dockerfile
+FROM busybox
+WORKDIR /src
+COPY foo .
+```
+
+And this directory structure:
+
+```
+.
+├── Dockerfile
+├── bar
+├── foo
+└── node_modules
+```
+
+The legacy builder sends the entire directory to the daemon, including `bar`
+and `node_modules` directories, even though the `Dockerfile` does not use
+them. When using [BuildKit](../buildkit/index.md), the client only sends the
+files required by the `COPY` instructions, in this case `foo`.
+
+In some cases you may want to send the entire context:
+
+```dockerfile
+FROM busybox
+WORKDIR /src
+COPY . .
+```
+
+But exclude some files or directories. You can use a [`.dockerignore`](../../engine/reference/builder.md#dockerignore-file)
+file to exclude them from being sent like:
+
+```gitignore
+# .dockerignore
+node_modules
+bar
+```
+
+> **Warning**
+>
+> Avoid using your root directory, `/`, as the `PATH` for your build context,
+> as it causes the build to transfer the entire contents of your hard drive to
+> the daemon.
+{:.warning}
+
+## `URL` context
+
+The `URL` parameter can refer to three kinds of resources:
+* [Git repositories](#git-repositories)
+* Pre-packaged [tarball contexts](#tarball-contexts)
+* Plain [text files](#text-files)
+
+### Git repositories
+
+When the `URL` parameter points to the location of a Git repository, the
+repository acts as the build context. The builder recursively pulls the
+repository and its submodules. The pulls ignore the commit history. A
```

Comment:
"The pulls ignore the commit history"

Can we add some clarification for this? Is this just removing the `.git` directory, or is it a shallow-clone, or something else?